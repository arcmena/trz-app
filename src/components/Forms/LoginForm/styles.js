import styled from 'styled-components';

import { VpnKey } from '../../../styles/Icons';

export const Form = styled.form``;

export const IdDiv = styled.div`
    display: flex;
    align-items: center;

    width: 320px;
    height: 60px;
    border: 2px solid rgba(48, 48, 48, 0.35);
    border-radius: 15px;

    padding: 8px 10px;
    margin-bottom: 30px;

    input {
        border: none;
        font-family: 'Ubuntu';
        font-size: 20px;
        color: #383838;
    }
`;

export const KeyIcon = styled(VpnKey)`
    width: 30px;
    height: 30px;
    color: rgba(48, 48, 48, 0.54);
`;

export const JoinButton = styled.button`
    border: none;
    border-radius: 15px;
    background: #61c1d4;
    width: 320px;
    height: 60px;

    span {
        font-family: 'Ubuntu';
        font-size: 30px;
        color: #ffffff;
    }

    cursor: pointer;
    transition: background 0.3s;

    &:hover {
        background: #55adbe;
    }

    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
`;
