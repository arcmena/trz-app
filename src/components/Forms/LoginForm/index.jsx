import React from 'react';

import { IdDiv, KeyIcon } from './styles';
import { Button } from '../../';

export default ({ onChange, onSubmit }) => {
    return (
        <form onChange={onChange} onSubmit={onSubmit}>
            <IdDiv>
                <input type="text" name="id" placeholder="Identification" />
                <KeyIcon />
            </IdDiv>

            <Button type="submit">
                <span>LOGIN</span>
            </Button>
        </form>
    );
};
