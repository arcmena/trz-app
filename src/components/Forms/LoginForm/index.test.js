import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './';
import { Button } from '../../';

const essentialProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
};

describe('<LoginForm/>', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<LoginForm {...essentialProps} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the form', () => {
        const wrapper = shallow(<LoginForm {...essentialProps} />);
        expect(wrapper.find('input').props()).toEqual({
            type: 'text',
            name: 'accessKey',
            placeholder: 'Identification',
        });
        expect(wrapper.find('KeyIcon'));
        expect(wrapper.find(Button).childAt(0).text()).toBe('LOGIN');
    });
    it('should call onChange when the value of the input changes', () => {
        const wrapper = shallow(<LoginForm {...essentialProps} />);
        essentialProps.onChange.mockReset();
        wrapper.find('input').simulate('change');
        expect(essentialProps.onChange);
    });
    it('should call onClick when the buttonis pressed', () => {
        const wrapper = shallow(<LoginForm {...essentialProps} />);
        essentialProps.onChange.mockReset();
        wrapper.find(Button).simulate('click');
        expect(essentialProps.onChange);
    });
});
