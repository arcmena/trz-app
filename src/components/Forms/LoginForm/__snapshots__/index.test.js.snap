// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> should render as expected 1`] = `
<form
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
>
  <styled.div>
    <input
      name="accessKey"
      placeholder="Identification"
      type="text"
    />
    <Styled(VpnKey) />
  </styled.div>
  <styled.button
    type="submit"
  >
    <span>
      LOGIN
    </span>
  </styled.button>
</form>
`;
